core/src/main/java/com/oath/oak/OakAttachedReadBuffer.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2018 Oath Inc.
+ * Licensed under the terms of the Apache 2.0 license.
+ * Please see LICENSE file in the project root for terms.
+ */
+
+package com.oath.oak;
+
+import sun.nio.ch.DirectBuffer;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+class OakAttachedReadBuffer implements OakReadBuffer, OakUnsafeDirectBuffer {
+
+    protected ByteBuffer bb;
```

Comment:
I remember we told: "OakReadBuffer inherits from Slice and can use Slice’s protected ByteBuffer for reading" and this is wrote in our design document...

Reply:
This design will be applied in the following PR.